### ISSUE TITLE
Notepack mongoose object problem

### ISSUE THREAD

**Issue by dorukgunes** [darrachequesne/notepack#5]
Do you know any elegant way to encode and mongoose object with notepack ? I get a RangeError: Maximum call stack size exceeded when i try to encode and mongoose object.

**Comment by darrachequesne** [comment]
Did you try [toJSON](http://mongoosejs.com/docs/guide.html#toJSON)? Or maybe with [Query#lean](http://mongoosejs.com/docs/api.html#query_Query-lean) when querying the documents?

**Comment by dorukgunes** [comment]
That works but in our project it is complicated to change our queries into lean, i am writing this about socket-io redis because in version 5 it uses notepack, before notepack there is no problem with encoding mongoose object. Dou you have any idea how i can change the encoder of socket-io.redis or come over this problem without changing every query.

**Comment by darrachequesne** [comment]
@dorukgunes and what about toJSON? It seems [msgpack-lite](https://github.com/kawanet/msgpack-lite) does not call toJSON() method when available, but [msgpack-js](https://github.com/creationix/msgpack-js) does ([there](https://github.com/creationix/msgpack-js/blob/0.3.0/msgpack.js#L349)):
```js
if('function' === typeof value.toJSON)
  return encode(value.toJSON(), buffer, offset)
```

**Comment by darrachequesne** [comment]
Closed by #8
